### ISSUE TITLE
Improve exception messages, handling, and logging when channel task errors occur

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3300]
Imported Issue. Original Details:
Jira Issue Key: MIRTH-3399
Reporter: wayneh
Created: 2014-08-11T19:32:25.000-0700

**Comment by rbeckman-nextgen** [comment]
Improved the exception messages, handling, and logging when channel task errors occur. StartException, StopException, etc now only occur at the channel level instead of at the connector levels too. This prevents an exception from being wrapped multiple times in a StopException (and similarly for the other types). Connector tasks now always throw a ConnectorTaskException regardless of the task type since the type can be inferred from the Channel level exception and stack trace. The messages displayed when halting channels has been improved as well.

Imported Comment. Original Details:
Author: wayneh
Created: 2014-08-11T20:07:40.000-0700

**Comment by rbeckman-nextgen** [comment]
Created some simple channels for testing in 3.0.3, forcing it to error out and log statements during certain states, Deploy/Undeploy, Start/Stop, etc... I moved these channels into 3.1 and also caused the same errors to occur. The errors were inspected side-by-side, for comparison sake, to ensure the exception messages were updated/clear. Also looked through the code so as to get a handle of the internals.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-03T15:23:31.000-0700
